src/server/bg_services/log_replication_scanner.js

```diff
@@ -156,6 +155,15 @@ class LogReplicationScanner {
         }
     }
 
+    async process_copy_candidate(src_bucket, dst_bucket, candidate) {
+        const src_object_info = candidate.src_object_info;
+        const dst_object_info = candidate.dst_object_info;
+        if (src_object_info && (!dst_object_info || (src_object_info.LastModified > dst_object_info.LastModified ||
+            src_object_info.ETag !== dst_object_info.ETag))) {
```

Comment:
@dannyzaken I think, as per @utkarsh-pro comment [here](https://github.com/noobaa/noobaa-core/pull/7349#discussion_r1236490042) there is case due to log lag where `src_obj` is older than `dst_obj` but the object is not update in dst, so we will not consider that case in this logic.

Reply:
@achouhan09 I referred to it in [this comment](https://github.com/noobaa/noobaa-core/pull/7349#discussion_r1245275477) - since we treat the log entries as a hint to look at a specific object, then once we copied from source to dest, we should have the eTags matching. If the eTags do not match and the dest date is newer, it can only mean that the object was uploaded directly to the dest more recently than the source object was uploaded. In that case, we want to skip it. 
hope it makes sense 🙃

Reply:
@utkarsh-pro @romayalon would be happy to get your opinion also

Reply:
Make sense to me now. Thanks👍